src/Symfony/Component/Form/Exception/TransformationFailedException.php

```diff
@@ -18,4 +18,35 @@
  */
 class TransformationFailedException extends RuntimeException
 {
+    private $messageKey;
+    private $messageData;
+
+    public function __construct($message = '', $code = 0, \Exception $previous = null, $messageKey = null, $messageData = [])
+    {
+        parent::__construct($message, $code, $previous);
+
+        $this->setSafeMessage($messageKey, $messageData);
+    }
+
+    /**
+     * Set a message that will be shown to the user.
+     *
+     * @param string $messageKey  The message or message key
+     * @param array  $messageData Data to be passed into the translator
+     */
+    public function setSafeMessage($messageKey, array $messageData = array())
```

Comment:
I hesitated, because it's kind of boring to pass the status code and previous exception in order to set it...but an extra assignment isn't really better, though.

Making this method fluent might be an option in order to write something like:

```php
throw (new TransformationFailedException('Expected numeric value'))->setSafeMessage('Money amount should be numeric. {{ amount }} is invalid.', array('{{ amount }}' => json_encode($amount)));
```